polkadot/node/core/candidate-validation/src/lib.rs

```diff
@@ -773,7 +837,37 @@ async fn validate_candidate_exhaustive(
 					// invalid.
 					Ok(ValidationResult::Invalid(InvalidCandidate::CommitmentsHashMismatch))
 				} else {
-					Ok(ValidationResult::Valid(outputs, (*persisted_validation_data).clone()))
+					let core_index = candidate_receipt.descriptor.core_index();
+
+					match (core_index, exec_kind) {
+						// Core selectors are optional for V2 descriptors, but we still check the
```

Comment:
the core selector is optional for now because we don't want to break existing parachains.
AFAIK the idea is to make it mandatory in the future

Reply:
> Hijacking the comment because `CoreSelector`s are mentioned here, but this is also a remark to past PRs.
> 
> `CoreSelector`s are effectively only used when a parachain is wanting to enable elastic scaling, correct?
> 

Yes, mandatory only if the parachain wants to use elastic scaling.

> When a parachain has 3 parablocks per relaychain block the `CoreSelector` will essentially inform the RLC which of the 3 cores the parablock belongs to. So for 3 cores we only ever expect values of 1-3 for the core selector as it is making only the decision between the already assigned to that para cores.

The core selector is a sequence number (the LSB of parachain blocknumber in practice). It can be used to determine the core index given a relay parent and the core indices assigned to the parachain. 

> 
> I think we should rename `CoreSelector` to `AssignedCoreSelector` or `ElasticCoreSelector` to also reflect that this is the index only of the assigned cores/cores used in elastic scaling and not all cores. This might prevent some weird assumptions or mistakes in the future.

I think the current name is fine given the context of usage.